assets/services/assets-api/src/utils/number.ts

```diff
@@ -95,3 +95,30 @@ export const addDecimalsAndConvertToHex = (number, decimals): string => {
     throw new Error(`addDecimalsAndConvertToHex --> ${error.message}`);
   }
 };
+
+/**
+ * [Add Numbers By Converting Into BigNumber]
+ *
+ * The purpose of this function is to add numbers after converting
+ * them to BigNumber and then return the Number
+ *
+ * This function is used everytime we add two numbers.
+ *
+ */
+export const addNumbersByConvertingIntoBigNumber = (number1, number2): number => {
```

Comment:
If number1 and number2 is already converted to float types, we might still get the wrong answer here.

Imagine a token with 18 decimal places using uint256 on solidity side. We can surly end up in a situation where number1 and number2 is unable to accurately capture this, which means converting them to BigNumber later is too late to fully solve the problem.

Reply:
That thing is already taken care before, they use this function - removeDecimalsFromBalances everytime they read from the blockchain. Then convert from BigNumber to Number and return
But they missed the case when adding numbers, they did it directly. For this also we need to convert first into BigNumber and then return it as normal number